yt_dlp/utils.py

```diff
@@ -5424,6 +5424,7 @@ def jwt_encode_hs256(payload_data, key, headers={}):
 # can be extended in future to verify the signature and parse header and return the algorithm used if it's not HS256
 def jwt_decode_hs256(jwt):
     header_b64, payload_b64, signature_b64 = jwt.split('.')
+    payload_b64 += (4 - (len(payload_b64) % 4)) * '='
     payload_data = json.loads(base64.urlsafe_b64decode(payload_b64))
```

Comment:
```suggestion
    payload_data = json.loads(base64.urlsafe_b64decode(f'{payload_b64}==='))
```
should work afair

Reply:
I does not work for me.
```$ ./yt-dlp.sh -Jv 'https://podbay.fm/p/behind-the-bastards'
[debug] Command-line config: ['-Jv', 'https://podbay.fm/p/behind-the-bastards']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.09.01 [5d7c7d656] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 4d8f5e2cb
[debug] Python 3.9.13 (CPython 64bit) - Linux-5.15.62-x86_64-with-glibc2.34 (glibc 2.34)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1 (setts), ffprobe 5.0.1
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1674 extractors
[debug] [PodbayFMChannel] Extracting URL: https://podbay.fm/p/behind-the-bastards
[PodbayFMChannel] behind-the-bastards: Downloading JSON metadata
[download] Downloading playlist: Behind the Bastards
ERROR: Incorrect padding
Traceback (most recent call last):
  File "/home/schnusch/Desktop/yt-dlp/yt_dlp/YoutubeDL.py", line 1477, in wrapper
    return func(self, *args, **kwargs)
  File "/home/schnusch/Desktop/yt-dlp/yt_dlp/utils.py", line 2992, in <lambda>
    return type(self.ydl)._handle_extraction_exceptions(lambda _, i: self._entries[i])(self.ydl, i)
  File "/home/schnusch/Desktop/yt-dlp/yt_dlp/utils.py", line 2827, in __getitem__
    entries = self.getslice(idx, idx + 1)
  File "/home/schnusch/Desktop/yt-dlp/yt_dlp/utils.py", line 2818, in getslice
    return list(self._getslice(start, end))
  File "/home/schnusch/Desktop/yt-dlp/yt_dlp/utils.py", line 2853, in _getslice
    page_results = self.getpage(pagenum)
  File "/home/schnusch/Desktop/yt-dlp/yt_dlp/utils.py", line 2812, in getpage
    page_results = [] if pagenum > self._pagecount else list(self._pagefunc(pagenum))
  File "/home/schnusch/Desktop/yt-dlp/yt_dlp/extractor/podbayfm.py", line 74, in <lambda>
    self._results_from_page(
  File "/home/schnusch/Desktop/yt-dlp/yt_dlp/extractor/podbayfm.py", line 60, in _results_from_page
    return [{
  File "/home/schnusch/Desktop/yt-dlp/yt_dlp/extractor/podbayfm.py", line 61, in <listcomp>
    **result_from_props(e),
  File "/home/schnusch/Desktop/yt-dlp/yt_dlp/extractor/podbayfm.py", line 7, in result_from_props
    thumbnail = jwt_decode_hs256(image_jwt).get('url') if image_jwt else None
  File "/home/schnusch/Desktop/yt-dlp/yt_dlp/utils.py", line 5428, in jwt_decode_hs256
    payload_data = json.loads(f'{base64.urlsafe_b64decode(payload_b64)}===')
  File "/nix/store/pn7863n7s2p66b0gazcylm6cccdwpzaf-python3-3.9.13/lib/python3.9/base64.py", line 133, in urlsafe_b64decode
    return b64decode(s)
  File "/nix/store/pn7863n7s2p66b0gazcylm6cccdwpzaf-python3-3.9.13/lib/python3.9/base64.py", line 87, in b64decode
    return binascii.a2b_base64(s)
binascii.Error: Incorrect padding```

Reply:
the line in your log:
```py
payload_data = json.loads(f'{base64.urlsafe_b64decode(payload_b64)}===')
```
is not the same as the line that was suggested:
```py
payload_data = json.loads(base64.urlsafe_b64decode(f'{payload_b64}==='))
```

Reply:
You're right.